interview-questions.md

```diff
@@ -8,62 +8,66 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer: A parameter is defined by a function and is found within parentheses after the function name. An argument affects change within the function. 
 
+  Researched answer: Where the function is invoked, arguments are found in parentheses after the function name, and serve to pass content to the function. For every argument, there is an equal number of paremeters. Arguments can be different data types such as an array of numbers, or strings. 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: The predefined paremeters taken by the .map method are values that lie within an array. I do not recall which parameters are required and which are optional, but I will gladly do more research on this method and follow up with my findings. 
 
+  Researched answer: The predefined parameters of the .map() built-in method are value, index, and array. The value is required, whereas the index and array are optional. 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Both map and filter are built-in methods for arrays. Map is used to return an array with the same number of elements (or values), whether altered or not, and filter is used to filter out certain elements (or values), returning an array containing fewer than it originated with. 
 
+  Researched answer: The map method uses iteration through an array, and will return an updated array of the same length. The filter method also iterates through an array however, this method will only return an array containing values that meet a specified condition. 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: A function is code that works like a mini machine to execute actions defined by the developer. A method is a built-in function with a predefined role that can be called upon when needed, and can be used within a function. 
 
+  Researched answer: A function defines a task, or set of tasks that can be executed over and over following detailed instructions. A method performs tasks similarly to a function however, a method is associated with an object while a function is not. 
 
 
 5. What is object destructuring?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: Destructuring assigns values individually within an array. 
 
+  Researched answer: Destructuring allows for the existence of individual variables within the assignment of a single array. For example, an array containing information about cars may include make, model, and year. We could log these variables separately, with different values each time. 
```

Comment:
Make sure to read the question thoroughly.  Your answer is correct for array destructuring, and object destructuring is basically the same except it operates on objects not arrays.